kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -719,6 +726,73 @@ impl<'a, 'b> CrateAnalyzer<'a, 'b> {
         Ok(())
     }
 
+    /// Emit the Kythe edges for cross references for imports in this crate
+    pub fn emit_import_xrefs(&mut self) -> Result<(), KytheError> {
+        // We must clone to avoid double borrowing "self"
+        let analysis = self.krate.analysis.clone();
+
+        let mut template_vname = self.krate_vname.clone();
+        template_vname.set_language("rust".to_string());
+        let krate_signature = template_vname.get_signature();
+
+        for reference in &analysis.imports {
+            // If there is no reference id, we can't emit a cross reference
+            if reference.ref_id.is_none() {
+                continue;
+            }
+
+            // Currently this is only known to support `Use` ImportKinds
+            if reference.kind != rls_data::ImportKind::Use {
+                continue;
+            }
+
+            let ref_id = reference.ref_id.unwrap();
+
+            let mut reference_vname = template_vname.clone();
+            let span = &reference.span;
+
+            // Create VName for target of reference
+            let mut target_vname = template_vname.clone();
+            target_vname.set_signature(format!("{}_def_{}", krate_signature, ref_id.index));
+            target_vname.set_language("rust".to_string());
+
+            // Create VName for the reference node
+            let file_vname = self.file_vnames.get(span.file_name.to_str().unwrap());
```

Comment:
maybe a helper function to do all the to_str and unwrapping? this is a pretty common stanza.